hooks/shapediver/viewer/useInteraction.ts

```diff
@@ -0,0 +1,159 @@
+import { useEffect, useRef, useState } from "react";
+import { useShapeDiverStoreViewer } from "../../../store/useShapeDiverStoreViewer";
+import { HoverManager, InteractionEngine, MultiSelectManager, SelectManager } from "@shapediver/viewer.features.interaction";
+import { MaterialStandardData, sceneTree } from "@shapediver/viewer";
+import { vec3 } from "gl-matrix";
+
+/**
+ * Hook allowing to create the interaction engine and the managers that are specified via the settings.
+ * 
+ * @param viewportId 
+ */
+export function useInteraction(viewportId: string, settings?: {
+	interactionTypes?: {
+		drag?: boolean,
+		select?: boolean,
+		hover?: boolean,
+	},
+	multiple?: boolean,
+	output?: string | string[],
+	pattern?: string | string[],
+	groupNodes?: boolean,
+	selectedNodes?: {
+		node?: {
+			output: string,
+			pattern?: string,
+			nodeName: string,
+			path: string
+		},
+		nodes?: {
+			output: string,
+			pattern?: string,
+			nodeName: string,
+			path: string
+		}[]
+	}
+}): {
```

Comment:
@MajorMeerkatThe3rd I suggest to type the settings as an interface, export it, and use it for the App Builder widget as well